### PR TITLE
Wysiwyg markdown

### PR DESCRIPTION
**Pull request by ronronscelestes**
# Wysiwyg markdown actions added + tests

## Description
Main markdowns (bold, italic, etc)
List markdowns (bullet, number)
Title markdowns (h1, h2, h3, etc)

### CODE REVIEW

File: example/src/shared/Wysiwyg/_test_/Wysiwyg.spec.js [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    fireEvent.click(getByText("h1"));
+    fireEvent.mouseDown(container.querySelector("#selectTitle"));
+    fireEvent.click(getByText("h4"));
+    fireEvent.mouseDown(container.querySelector("#selectTitle"));
+    fireEvent.click(getByText("h1"));
+
+    expect(queryByText("####")).not.toBeInTheDocument();
+    expect(getByText("#")).toBeInTheDocument();
+  });
+});
```
Comment (HichamELBSI) [inline example/src/shared/Wysiwyg/_test_/Wysiwyg.spec.js:279]: Super cool tests 😄 . If you want to do something before each test like the `render(...)`, you can call the `beforeEach()` function.

```js
describe("Wysiwyg", () => {
    let renderedContainer;
    let getSomethingByText;
    let querySomethingByText;

    beforeEach(() => {
        const { container, getByText, queryByText } = render(
            <ThemeProvider theme={lightTheme}>
               <Wysiwyg label={"hello world"} placeholder={""} onChange={jest.fn()} />
            </ThemeProvider>
         );
         renderedContainer = container;
         getContainerByText = getByText;
         containerQueryByText = queryByText;
    });

    // In your tests
    it("should ...",  () => {
         await waitFor(() => renderedContainer.querySelector(".CodeMirror-cursor"));
         expect(getContainerByText("hello world")).toBeInTheDocument();
         // ...;
    });
});
```

File: example/src/shared/Wysiwyg/_test_/Wysiwyg.spec.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
+import * as React from "react";
+import { render, waitFor, prettyDOM, fireEvent } from "@testing-library/react";
```
Comment (HichamELBSI) [inline example/src/shared/Wysiwyg/_test_/Wysiwyg.spec.js:2]: Do you still need prettyDOM ?

**Review by HichamELBSI — COMMENTED** [review]
Good job @ronronscelestes, So impressed by the work 👏🏽

File: example/src/shared/Wysiwyg/_test_/Wysiwyg.spec.js (hunk 2 above)
Comment (ronronscelestes) [reply example/src/shared/Wysiwyg/_test_/Wysiwyg.spec.js:2]: Oh no you're right, will remove it!

File: example/src/shared/Wysiwyg/_test_/Wysiwyg.spec.js (hunk 1 above)
Comment (ronronscelestes) [reply example/src/shared/Wysiwyg/_test_/Wysiwyg.spec.js:279]: nice tip thank you, will try it!

File: example/src/shared/Wysiwyg/index.js [hunk 3]
```diff
@@ -1,31 +1,74 @@
...
+  return (
+    <>
+      <TextButton>{label}</TextButton>
+      <Box paddingTop={1}>
+        <WysiwygNav 
+          placeholder={placeholder} 
+          onActionClick={handleActionClick}
+        />
+        <Editor 
+          // value={value || ''}
```
Comment (mfrachet) [inline example/src/shared/Wysiwyg/index.js:56]: You can remove this one too 😛

File: example/src/shared/Wysiwyg/index.js [hunk 4]
```diff
@@ -1,31 +1,74 @@
...
+    <>
+      <TextButton>{label}</TextButton>
+      <Box paddingTop={1}>
+        <WysiwygNav 
+          placeholder={placeholder} 
+          onActionClick={handleActionClick}
+        />
+        <Editor 
+          // value={value || ''} 
+          onChange={modifiedData => onChange(modifiedData)}
```
Comment (mfrachet) [inline example/src/shared/Wysiwyg/index.js:57]: You can factor this by removing the parenthesis from the arrow function and the underlying function

```suggestion
          onChange={onChange}
```

File: example/src/shared/Wysiwyg/index.js [hunk 5]
```diff
@@ -1,31 +1,74 @@
...
+  );
 };
 
 Wysiwyg.propTypes = {
   label: PropTypes.string,
-  name: PropTypes.string,
   onChange: PropTypes.func,
   placeholder: PropTypes.string,
-  value: PropTypes.string,
+  // value: PropTypes.string,
```
Comment (mfrachet) [inline example/src/shared/Wysiwyg/index.js:71]: You can also remove this one I think 😛

**Pull request merged**